### PR TITLE
MOSIP-25804 Fixed automation failure.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentOtpController.java [hunk 1]
```diff
@@ -71,9 +71,11 @@ public IndividualIdResponseDto reqOtpForIndividualId(@RequestBody IndividualIdOt
 		audit.setAuditRequestDto(EventEnum.OTP_INDIVIDUALID_GEN);
 		IndividualIdResponseDto individualIdResponseDto;
 		try {
-		if(individualIdRequestDto.getIndividualId()  == null) {
-			throw new ResidentServiceCheckedException(ResidentErrorCode.INVALID_INPUT.getErrorCode(), ResidentErrorCode.INVALID_INPUT.getErrorMessage() + "individualId");
-		}
+			if(individualIdRequestDto.getIndividualId()  == null || individualIdRequestDto.getIndividualId().equalsIgnoreCase("")
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentOtpController.java]: Replace equalsIgnoreCase with equals

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java [hunk 2]
```diff
@@ -475,7 +475,14 @@ public Tuple2<ResponseWrapper<VidRevokeResponseDTO>, String> revokeVidV2(BaseVid
 				}
 			}
 		} else {
-			String uinForVid = identityServiceImpl.getUinForIndividualId(vid);
+			String uinForVid ="";
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java]: No need to initialize with ""

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 3]
```diff
@@ -1090,7 +1091,7 @@ public void validateDownloadCardRequest(MainRequestDTO<DownloadCardRequestDTO> d
 	}
 
 	private void validateIndividualIdV2(String individualId) {
-		if (individualId == null || StringUtils.isEmpty(individualId)) {
+		if (individualId == null || StringUtils.isEmpty(individualId) || isNumeric(individualId)) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: We should not assume that individual ID is always numeric. If required use the UIN validator/VID validator/RID validator

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java]: done.

File: resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentOtpController.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentOtpController.java]: done.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: replace with rid validator.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**